docs/API.md

```diff
@@ -0,0 +1,203 @@
+    *   Returns the name of the user by Twitter ID (note that Twitter ID different from Twitter handle) **at the time the name was claimed.**
+    
+    ```plain
+    {endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "name_by_twitter_id": { "twitter_id": {twitter_id} }})}
+    ```
+    
+*   Execute Message - claim
+*   Execute Message - update\_verifier\_pub\_keys
+*   Execute Message - set\_verification\_threshold
+*   Execute Message - set\_name\_nft\_address
+*   Execute Message - set\_minting\_fee
+*   Execute Message - withdraw\_funds
+
+### **Name NFT API**
+
+*   Query - admin
+    *   Returns the admin address
+*   Query - is\_admin
+    *   Checks whether the specified address is an admin
+*   Query - transferrable
+    *   Returns whether Name NFTs are currently transferrable
+*   Query - owner\_of
+    *   Check the owner of the given token ID (optional filter on including expired owners)
+*   Query - approval
+    *   Check for the approval response for a given token ID and spender (optional filter on including expired owners)
+*   Query - approvals
+    *   Check for the approval response for a given token ID (optional filter on including expired owners)
+*   Query - all\_operators
+*   Query - num\_tokens
+*   Query - contract\_info
+*   Query - nft\_info
+*   Query - all\_nft\_info
+*   Query - tokens
+*   Query - all\_tokens
+*   Query - minter
+
+Examples
+--------
+
+**Get a bech32 address from an ICNS name**
+
+```plain
+{endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "address-by-icns": { "icns": {twitter_username}.{bech32_prefix} }})}
+```
+
+*   Example: [https://lcd.testnet.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJpY25zX25hbWVzIjogeyJhZGRyZXNzIjogIm9zbW8xdnY2aHJ1cXV6cHR5NHhwa3M5em5rdzhneXM1eDRuc25nNGtlcnkifX0K](https://lcd.testnet.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJpY25zX25hbWVzIjogeyJhZGRyZXNzIjogIm9zbW8xdnY2aHJ1cXV6cHR5NHhwa3M5em5rdzhneXM1eDRuc25nNGtlcnkifX0K)
+*   Response
+
+```plain
+{
+  "data": {"names":["TonyYun9.osmo"],"primary_name":"TonyYun9.osmo"}
+}
+```
+
+  
+
+**Get a ICNS name of a bech32 address**
+
+```plain
+{endpoint}/cosmwasm/v1/contract/{resolver_contract_address}/smart/{base64Encoding({ "address-by-icns": { "icns": {twitter_username}.{bech32_prefix} }})}
+```
+
+*   Example: [https://lcd.testnet.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJhZGRyZXNzIjogeyJuYW1lIjogIlRvbnlZdW45IiwgImJlY2gzMl9wcmVmaXgiOiAib3NtbyJ9fQo=](https://lcd.testnet.osmosis.zone/cosmwasm/wasm/v1/contract/osmo1qjs29tsavxuk39dugs7lny7cc28a57hqa9u9q0ysr2qat96aghrqzgp60x/smart/eyJhZGRyZXNzIjogeyJuYW1lIjogIlRvbnlZdW45IiwgImJlY2gzMl9wcmVmaXgiOiAib3NtbyJ9fQo=)
+*   Response
+
+```plain
+{
+  "data": {"address":"osmo1vv6hruquzpty4xpks9znkw8gys5x4nsng4kery"}
```

Comment:
Updating the response to reflect the new example in https://github.com/icns-xyz/icns/pull/19#discussion_r1089647680
```suggestion
  "data": {"names": ["NosNode_", "NosNode_Intern"], "primary_name": "NosNode_"}
```